interfaces/builtin/camera_test.go

```diff
@@ -0,0 +1,99 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin_test
+
+import (
+	. "gopkg.in/check.v1"
+
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/interfaces/apparmor"
+	"github.com/snapcore/snapd/interfaces/builtin"
+	"github.com/snapcore/snapd/snap"
+	"github.com/snapcore/snapd/testutil"
+)
+
+type CameraInterfaceSuite struct {
+	iface interfaces.Interface
+	slot  *interfaces.Slot
+	plug  *interfaces.Plug
+}
+
+var _ = Suite(&CameraInterfaceSuite{
+	iface: builtin.MustInterface("camera"),
+})
+
+const cameraConsumerYaml = `name: consumer
+apps:
+ app:
+  plugs: [camera]
+`
+
+const cameraCoreYaml = `name: core
+type: os
+slots:
+  camera:
+`
+
+func (s *CameraInterfaceSuite) SetUpTest(c *C) {
+	s.plug = MockPlug(c, cameraConsumerYaml, nil, "camera")
+	s.slot = MockSlot(c, cameraCoreYaml, nil, "camera")
+}
+
+func (s *CameraInterfaceSuite) TestName(c *C) {
+	c.Assert(s.iface.Name(), Equals, "camera")
+}
+
+func (s *CameraInterfaceSuite) TestSanitizeSlot(c *C) {
+	c.Assert(s.slot.Sanitize(s.iface), IsNil)
+	slot := &interfaces.Slot{SlotInfo: &snap.SlotInfo{
+		Snap:      &snap.Info{SuggestedName: "core"},
```

Comment:
Thanks for the findings. I totally agree here.
I had a quick walk-through the interfaces I recently worked on for the improvement, I only found this minor issue in camera_test. For other test cases I worked on, we're now using `Suggestedname: "some-snap"` to differentiate from `core`. It seems to me it's a legacy issue since there are a few more interfaces which are not in my todo list still using this pattern `SuggestedName: "core"`. 
I'll fix it in camera_test in next commit.
It'd best we could resolve this problem for the rest in a separated PR.